src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -315,313 +347,326 @@ void func_80B67540(EnSth* this, PlayState* play) {
+    } else if (EnSth_CanSpeakToPlayer(this, play) || this->actor.isTargeted) {
         if ((gSaveContext.save.time >= CLOCK_TIME(6, 0)) && (gSaveContext.save.time <= CLOCK_TIME(18, 0))) {
-            this->actor.textId = 0x1130;
+            this->actor.textId = 0x1130; // Huh? The Moon...
         } else {
-            this->actor.textId = 0x1131;
+            this->actor.textId = 0x1131; // (The Moon) gotten bigger again
         }
         func_800B8614(&this->actor, play, 110.0f);
     }
-    this->unk_294.x = -0x1388;
+    this->headRot.x = -0x1388;
 }
 
-void func_80B67958(EnSth* this, PlayState* play) {
+// used by type: STH_TYPE_UNUSED_1 and undefined types
+void EnSth_DefaultIdle(EnSth* this, PlayState* play) {
     SkelAnime_Update(&this->skelAnime);
 }
 
-void func_80B67984(EnSth* this, PlayState* play) {
-    u16 sp1E;
+void EnSth_GetInitialSwampSpiderHouseText(EnSth* this, PlayState* play) {
+    u16 nextTextId;
 
     if (CHECK_WEEKEVENTREG(WEEKEVENTREG_34_10)) {
-        sp1E = 0x903;
-        func_80B670A4(this, 2);
+        nextTextId = 0x903; // (does not exist)
+        EnSth_ChangeAnim(this, STH_ANIM_TALK);
     } else if (CHECK_WEEKEVENTREG(WEEKEVENTREG_34_20)) {
-        sp1E = 0x90F;
-        func_80B670A4(this, 2);
-    } else if (CHECK_WEEKEVENTREG(WEEKEVENTREG_34_40)) {
-        if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_34_08)) {
-            sp1E = 0x91B;
+        nextTextId = 0x90F; // (does not exist)
+        EnSth_ChangeAnim(this, STH_ANIM_TALK);
+    } else if (CHECK_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_MASK_OF_TRUTH)) {
+        if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_SWAMP_SPIDERHOUSE_TALKED)) {
+            nextTextId = 0x91B; // As soon as I calm down, getting rid of it
         } else {
-            sp1E = 0x918;
+            nextTextId = 0x918; // I've had enough of this, going home
         }
-        func_80B670A4(this, 2);
-    } else if (Inventory_GetSkullTokenCount(play->sceneId) >= 30) {
+        EnSth_ChangeAnim(this, STH_ANIM_TALK);
+    } else if (Inventory_GetSkullTokenCount(play->sceneId) >= SPIDER_HOUSE_TOKENS_REQUIRED) {
         if (INV_CONTENT(ITEM_MASK_TRUTH) == ITEM_MASK_TRUTH) {
-            this->unk_29C |= 4;
-            sp1E = 0x919;
+            this->sthFlags |= STH_FLAG_SWAMP_SPIDER_HOUSE_SAVED;
+            nextTextId = 0x919; // I've been saved!
         } else {
-            sp1E = 0x916;
+            nextTextId = 0x916; // I've been saved! (Duplicate)
         }
     } else if (CHECK_WEEKEVENTREG(WEEKEVENTREG_34_02)) {
-        sp1E = 0x8FF;
+        nextTextId = 0x8FF; // (does not exist)
     } else {
-        sp1E = 0x8FC;
+        nextTextId = 0x8FC; // (does not exist)
         SET_WEEKEVENTREG(WEEKEVENTREG_34_02);
     }
-    Message_StartTextbox(play, sp1E, &this->actor);
+
+    Message_StartTextbox(play, nextTextId, &this->actor);
 }
 
-void func_80B67AB4(EnSth* this, PlayState* play) {
+void EnSth_PostSwampSpiderHouseGiveMask(EnSth* this, PlayState* play) {
```

Comment:
```suggestion
void EnSth_TalkAfterSwampSpiderHouseGiveMask(EnSth* this, PlayState* play) {
```
Just an idea? You don't have to take it though